packages/web/src/dev/ValidatorsList.tsx

```diff
@@ -0,0 +1,623 @@
+import { BigNumber } from 'bignumber.js'
+import * as React from 'react'
+import { StyleSheet, Text as RNText, View } from 'react-native'
+import CopyToClipboard from 'src/dev/CopyToClipboard'
+import ProgressCutBar from 'src/dev/ProgressCutBar'
+import { H1 } from 'src/fonts/Fonts'
+import { I18nProps, withNamespaces } from 'src/i18n'
+import Chevron, { Direction } from 'src/icons/chevron'
+import { HEADER_HEIGHT } from 'src/shared/Styles'
+import { colors, standardStyles, textStyles, typeFaces } from 'src/styles'
+import { cutAddress, formatNumber, weiToDecimal } from 'src/utils/utils'
+
+class Text extends RNText {
+  render() {
+    return <RNText style={[styles.defaultText, this.props.style]}>{this.props.children}</RNText>
+  }
+}
+
+interface HeaderCellProps {
+  style: any[]
+  name: string
+  order: boolean | null
+  onClick: () => void
+}
+
+const HeaderCell = React.memo(function HeaderCellFn({
+  style,
+  name,
+  order,
+  onClick,
+}: HeaderCellProps) {
+  return (
+    <View onClick={onClick} style={[styles.tableHeaderCell, ...((style || []) as any)]}>
+      <Text>{name}</Text>
+      <Text
+        style={[
+          styles.tableHeaderCellArrow,
+          ...(order !== null ? [styles.tableHeaderCellArrowVisible] : []),
+        ]}
+      >
+        <Chevron direction={order ? Direction.up : Direction.down} color={colors.white} size={10} />
+      </Text>
+    </View>
+  )
+})
+
+interface ValidatorsListProps {
+  data: any
+  isLoading: boolean
+}
+
+export interface State {
+  expanded: number | undefined
+  orderBy: string
+  orderAsc: boolean
+}
+
+class ValidatorsListApp extends React.PureComponent<ValidatorsListProps & I18nProps, State> {
+  state = {
+    expanded: undefined,
+    orderBy: 'name',
+    orderAsc: true,
+  }
+  private orderAccessors = {
+    name: (_) => _.name.toLowerCase(),
+    total: (_) => _.numMembers * 1000 + _.elected,
+    votes: (_) => +_.votesAbsolute,
+    gold: (_) => _.gold,
+    commision: (_) => _.commission,
+    rewards: (_) => _.rewards,
+    uptime: (_) => _.uptime,
+  }
+  private defaultOrderAccessor = 'name'
+  private cachedCleanData
+
+  expand(expanded: number) {
+    if (this.state.expanded === expanded) {
+      this.setState({ expanded: undefined })
+    } else {
+      this.setState({ expanded })
+    }
+  }
+
+  orderBy(orderBy: string) {
```

Comment:
one small style nit.

there was a decision at one point for methods on react components classes to be defined like `expand = () => {}` this way they are always scoped to class.

Reply:
Can you give me an example for that?